src/Classes/Particle/Connecter.ts

```diff
@@ -0,0 +1,98 @@
+import {Particle} from "../Particle";
+import {Container} from "../Container";
+import {IRgb} from "../../Interfaces/IRgb";
+import {Utils} from "../Utils/Utils";
+
+export class Connecter {
+    private readonly particle: Particle;
+    private readonly container: Container;
+
+    constructor(container: Container, particle: Particle) {
+        this.container = container;
+        this.particle = particle;
+    }
+
+    /**
+     * Connecting particles on hover interactivity
+     */
+    public connect(destParticle: Particle) {
+        const container = this.container;
+        const options = container.options;
+        const particle = this.particle;
+
+        if (options.interactivity.events.onhover.enable && container.interactivity.status == 'mousemove') {
+            const xDiff = particle.position.x - destParticle.position.x;
+            const yDiff = particle.position.y - destParticle.position.y;
+            const mousePos = container.interactivity.mouse.position || {x: 0, y: 0};
+            const xCoreDiff = particle.position.x - mousePos.x;
+            const yCoreDiff = particle.position.y - mousePos.y;
+            const distMax = options.interactivity.modes.connect.distance;
+            const connectAreaRadius = options.interactivity.modes.connect.radius;
+
+            if ((xDiff < distMax && xDiff > -distMax)
+                && (yDiff < distMax && yDiff > -distMax)
+                && (xCoreDiff < connectAreaRadius && xCoreDiff > -connectAreaRadius)
+                && (yCoreDiff < connectAreaRadius && yCoreDiff > -connectAreaRadius)) {
+
+                const ctx = container.canvas.context;
+
+                if (!ctx) return;
+
+                const lineStyle = this.lineStyle(destParticle);
+
+                if (!lineStyle) return;
+
+                ctx.beginPath();
+                ctx.strokeStyle = lineStyle;
+                ctx.moveTo(particle.position.x, particle.position.y);
+                ctx.lineTo(destParticle.position.x, destParticle.position.y);
+                ctx.stroke();
+                ctx.closePath();
+            }
+
+        }
+    }
+
+    private static getStyleFromColor(color: IRgb): string {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of private static method not allowed after declaration of public instance method. Instead, this should come after constructors.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5098130)